src/borg/archive.py

```diff
@@ -882,6 +884,7 @@ def process_file(self, path, st, cache, ignore_inode=False):
                 cache.memorize_file(path_hash, st, [c.id for c in item.chunks])
             status = status or 'M'  # regular file, modified (if not 'A' already)
         item.update(self.stat_attrs(st, path))
+        item.size = sum(chunk.size for chunk in item.chunks)
```

Comment:
could also get size from st.st_size. risky due to race conditions? maybe it is safer like above.